engine/src/main/java/org/hibernate/validator/internal/engine/messageinterpolation/el/RootResolver.java

```diff
@@ -30,11 +28,9 @@
 	public static final String FORMATTER = "formatter";
 	private static final String FORMAT = "format";
 
-	private final Map<String, Object> map = Collections.synchronizedMap( new HashMap<String, Object>() );
-
 	@Override
 	public Class<?> getCommonPropertyType(ELContext context, Object base) {
-		return base == null ? String.class : null;
+		return null;
```

Comment:
Yes. In fact, it was already returning null before simplication as base was null when it was called.